### PR TITLE
Change validation error messages to include action name.

### PR DESCRIPTION
**Pull request by kunsingh**
This change is [Reviewable](https://reviewable.io/reviews/canoo/dolphin-platform/777)

### CODE REVIEW

File: platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java [hunk 1]
```diff
@@ -89,45 +82,42 @@ private boolean isFinal(Class<?> clazz) {
         return Modifier.isFinal(clazz.getModifiers());
     }
 
-    private boolean postConstructContainsParameter(Class<?> clazz) {
+    private void postConstructContainsParameter(Class<?> clazz) throws ControllerValidationException {
         List<Method> methods = ReflectionHelper.getInheritedDeclaredMethods(clazz);
         for (Method method : methods) {
             if (method.isAnnotationPresent(PostConstruct.class)) {
-                return checkParameterLength(method);
+                checkParameterLength(method, clazz);
```
Comment (hendrikebbers) [inline platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java]: Do the check here directly. It is just 1 if

File: platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java [hunk 2]
```diff
@@ -89,45 +82,42 @@ private boolean isFinal(Class<?> clazz) {
...
-        return false;
     }
 
-    private boolean preDestroyContainsParameter(Class<?> clazz) {
+    private void preDestroyContainsParameter(Class<?> clazz)  throws ControllerValidationException {
         List<Method> methods = ReflectionHelper.getInheritedDeclaredMethods(clazz);
         for (Method method : methods) {
             if (method.isAnnotationPresent(PreDestroy.class)) {
-                return checkParameterLength(method);
+                checkParameterLength(method, clazz);
```
Comment (hendrikebbers) [inline platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java]: Do the check here directly. It is just 1 if

File: platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java [hunk 3]
```diff
@@ -89,45 +82,42 @@ private boolean isFinal(Class<?> clazz) {
...
-        return false;
     }
 
-    private boolean checkParameterLength(Method method) {
+    private void checkParameterLength(Method method, Class<?> clazz) throws ControllerValidationException {
         if (method.getParameterTypes().length > 0) {
-            return true;
-        } else {
-            return false;
+            throw new ControllerValidationException("PostConstruct method "+ method.getName() +" should not contain parameter in Controller " + clazz.getName());
```
Comment (hendrikebbers) [inline platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java]: Text is wrong -> "PostCosntruct" but method called for PreDestroy, too. Just remove this method and do the check directly in "preDestroyContainsParameter"...

File: platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java [hunk 4]
```diff
@@ -89,45 +82,42 @@ private boolean isFinal(Class<?> clazz) {
         return Modifier.isFinal(clazz.getModifiers());
     }
 
-    private boolean postConstructContainsParameter(Class<?> clazz) {
+    private void postConstructContainsParameter(Class<?> clazz) throws ControllerValidationException {
```
Comment (hendrikebbers) [inline platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java]: rename method to check...

File: platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java [hunk 5]
```diff
@@ -89,45 +82,42 @@ private boolean isFinal(Class<?> clazz) {
...
             if (method.isAnnotationPresent(PostConstruct.class)) {
-                return checkParameterLength(method);
+                checkParameterLength(method, clazz);
             }
         }
-        return false;
     }
 
-    private boolean preDestroyContainsParameter(Class<?> clazz) {
+    private void preDestroyContainsParameter(Class<?> clazz)  throws ControllerValidationException {
```
Comment (hendrikebbers) [inline platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java]: rename method to check...

File: platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java [hunk 6]
```diff
@@ -89,45 +82,42 @@ private boolean isFinal(Class<?> clazz) {
...
         if (method.getParameterTypes().length > 0) {
-            return true;
-        } else {
-            return false;
+            throw new ControllerValidationException("PostConstruct method "+ method.getName() +" should not contain parameter in Controller " + clazz.getName());
         }
     }
 
-    private boolean isDolphinActionVoid(Class<?> clazz) {
+    private void checkDolphinActionVoid(Class<?> clazz) throws ControllerValidationException {
```
Comment (hendrikebbers) [inline platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java]: final

Comment (hendrikebbers) [reply platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java]: null check

File: platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java [hunk 7]
```diff
@@ -89,45 +82,42 @@ private boolean isFinal(Class<?> clazz) {
...
         List<Method> methods = ReflectionHelper.getInheritedDeclaredMethods(clazz);
```
Comment (hendrikebbers) [inline platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java]: final

File: platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java [hunk 8]
```diff
@@ -89,45 +82,42 @@ private boolean isFinal(Class<?> clazz) {
...
         for (Method method : methods) {
```
Comment (hendrikebbers) [inline platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java]: we have Java 8 now. Use Stream API ;)

File: platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java [hunk 9]
```diff
@@ -89,45 +82,42 @@ private boolean isFinal(Class<?> clazz) {
...
             if (method.isAnnotationPresent(DolphinAction.class)) {
-                return method.getReturnType().equals(Void.TYPE);
+                if(!method.getReturnType().equals(Void.TYPE)){
+                    throw new ControllerValidationException("DolphinAction "+ method.getName() + "must be void in Controller " + clazz.getName());
```
Comment (hendrikebbers) [inline platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java]: method.getName() is not enough. Name could be defined in @DolphinAction annotation ;)

Comment (hendrikebbers) [reply platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java]: "DolphinAction XYZ must be void in Controller XYZ" -> "Return type of controller action XYZ (method XYZ) in controller type XYZ (class XYZ) must be of type void"

Comment (hendrikebbers) [reply platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java]: clazz.getName() <- not enough. @DolphinController can define a name

File: platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java [hunk 10]
```diff
@@ -89,45 +82,42 @@ private boolean isFinal(Class<?> clazz) {
...
+                }
             }
         }
-        return true;
     }
 
-    private boolean isAnnotatedWithParam(Class<?> clazz) {
+    private void checkDolphinActionAnnotatedWithParam(Class<?> clazz)  throws ControllerValidationException {
         List<Method> methods = ReflectionHelper.getInheritedDeclaredMethods(clazz);
```
Comment (hendrikebbers) [inline platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java]: final

Comment (hendrikebbers) [reply platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java]: null check

File: platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java [hunk 11]
```diff
@@ -89,45 +82,42 @@ private boolean isFinal(Class<?> clazz) {
...
         for (Method method : methods) {
```
Comment (hendrikebbers) [inline platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java]: Stream API

File: platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java [hunk 12]
```diff
@@ -141,12 +131,11 @@ private boolean isAnnotatedWithParam(Class<?> clazz) {
                         }
                     }
                     if(!paramAnnotationFound) {
-                        return false;
+                        throw new ControllerValidationException("DolphinAction "+ method.getName() +" parameters must be annotated with @param in Controller " + clazz.getName());
```
Comment (hendrikebbers) [inline platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java]: see comments for the exception message

**Comment by hendrikebbers** [comment]
fix for https://github.com/canoo/dolphin-platform/issues/462

**Comment by hendrikebbers** [comment]
No additional unit tests needed here?

File: platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java (hunk 4 above)
Comment (kunsingh) [reply platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java]: Done.

File: platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java (hunk 1 above)
Comment (kunsingh) [reply platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java]: Done.

File: platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java (hunk 5 above)
Comment (kunsingh) [reply platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java]: Done.

File: platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java (hunk 2 above)
Comment (kunsingh) [reply platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java]: Done.

File: platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java (hunk 3 above)
Comment (kunsingh) [reply platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java]: Copy paste mistake....Done.

File: platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java (hunk 6 above)
Comment (kunsingh) [reply platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java]: Done.

File: platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java (hunk 8 above)
Comment (kunsingh) [reply platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java]: Within Lamda you can't throw checked exception.

File: platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java (hunk 9 above)
Comment (kunsingh) [reply platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java]: I have changed the message as suggested in above second point. Do I need to fetch name given with annotation?

File: platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java (hunk 10 above)
Comment (kunsingh) [reply platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java]: Done.

File: platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java (hunk 11 above)
Comment (kunsingh) [reply platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java]: Within Lamda you can't throw checked exception.

File: platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java (hunk 12 above)
Comment (kunsingh) [reply platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java]: Within Lamda you can't throw checked exception.

File: platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java [hunk 13]
```diff
@@ -51,145 +51,133 @@ public void validate(Class<?> clazz) throws ControllerValidationException {
...
     }
 
-    private boolean isFinal(Class<?> clazz) {
+    private boolean isFinal(final Class<?> clazz) {
         return Modifier.isFinal(clazz.getModifiers());
     }
 
-    private boolean postConstructContainsParameter(Class<?> clazz) {
+    private void checkPostConstructContainsParameter(final Class<?> clazz) throws ControllerValidationException {
         List<Method> methods = ReflectionHelper.getInheritedDeclaredMethods(clazz);
```
Comment (hendrikebbers) [inline platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java]: Assert.requeiesNotNull(clazz);
ReflectionHelper.getInheritedDeclaredMethods(clazz).filter(m -> method.isAnnotationPresent(PostConstruct.class)).filter(m -> method.getParameterTypes().length > 0).findAny()...

File: platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java [hunk 14]
```diff
@@ -51,145 +51,133 @@ public void validate(Class<?> clazz) throws ControllerValidationException {
...
 
-    private boolean isMoreThanOneDolphinModel(Class<?> clazz) {
+    private boolean isMoreThanOneDolphinModel(final Class<?> clazz) {
         List<Field> fields = ReflectionHelper.getInheritedDeclaredFields(clazz);
-        int count = 0;
-        for (Field field : fields) {
-            if (field.isAnnotationPresent(DolphinModel.class)) {
-                count++;
-            }
+        if(null != fields){
```
Comment (hendrikebbers) [inline platform/dolphin-platform-remoting-server/src/main/java/com/canoo/dp/impl/server/controller/ControllerValidator.java]: fields != null <- please use this as default order

**Comment by kunsingh** [comment]
Ready to merge

**Pull request merged**